### ISSUE TITLE
Authentication failed

### ISSUE THREAD

**Issue by dsgithub2k** [bbsan2k/plugin.video.f1tv#37]
Having issues lately using F1tv kodi add on . Unable to login with the Account Authentication failed. any help? I've checked the login and is exactly the one I use in other devices. Running Kodi on a Shield Pro.

**Comment by Calimerorulez** [comment]
Same here on Kodi on a Chromecast with Google TV...

**Comment by joeyberkovitz** [comment]
Seems that authentication API endpoint changed without any other significant changes. The fix for me was to make the following change in AccountManager.py
```
__ACCOUNT_API__='https://api.formula1.com/v2/account/'
__ACCOUNT_CREATE_SESSION__=__ACCOUNT_API__+'subscriber/authenticate/by-password'
```

**Comment by bbsan2k** [comment]
Interesting - so they finally deprecated v1 API... I‘ll try to provide the fix in an update tomorrow.
Thx @joeyberkovitz

**Comment by dsgithub2k** [comment]
> Seems that authentication API endpoint changed without any other significant changes. The fix for me was to make the following change in AccountManager.py
> 
> ```
> __ACCOUNT_API__='https://api.formula1.com/v2/account/'
> __ACCOUNT_CREATE_SESSION__=__ACCOUNT_API__+'subscriber/authenticate/by-password'
> ```

and how I go about changing that? or is it something only the developer can do it?

**Comment by dsgithub2k** [comment]
> Interesting - so they finally deprecated v1 API... I‘ll try to provide the fix in an update tomorrow.
> Thx @joeyberkovitz

That would be great!! especially we have Portuguese GP this weekend. Your app is fantastic by the way, thanks for doing this

**Comment by dsgithub2k** [comment]
> Same here on Kodi on a Chromecast with Google TV...

Hopefully we will get an update soon...

**Comment by bbsan2k** [comment]
Is anyone willing to try test release 0.1.5?
It fixes the Account Issue (on my platform - macOS, Kodi 18.8, Germany) but due to localized settings it may fail for others.
Please test and report!
Once this is done I'll deploy on the auto update repo.

**Comment by joeyberkovitz** [comment]
I verified that the test release works for me - windows and U.S.

**Comment by dsgithub2k** [comment]
> Is anyone willing to try test release 0.1.5?
> It fixes the Account Issue (on my platform - macOS, Kodi 18.8, Germany) but due to localized settings it may fail for others.
> Please test and report!
> Once this is done I'll deploy on the auto update repo.

I can### where do I download the 1.5 version?

**Comment by Calimerorulez** [comment]
I downloaded the 0.1.5 zip, but it says 0.1.4. It still crashes (in Holland), but I will experiment further.

**Comment by dsgithub2k** [comment]
Anywhere I can download the 0.1.5 zip here to test it?

**Comment by Calimerorulez** [comment]
> Anywhere I can download the 0.1.5 zip here to test it?

https://github.com/bbsan2k/plugin.video.f1tv/archive/0.1.5.zip

**Comment by joeyberkovitz** [comment]
@bbsan2k I think you forgot to bump the version on the test release, so the upgrade doesn't go through without an uninstall first.

**Comment by dsgithub2k** [comment]
uninstall current version, installed 0.1.5 test version. Authentication failed as well. Using Kodi on a Shield TV Pro. 
It does says the version is 0.1.4 as mentioned by @Calimerorulez

**Comment by bbsan2k** [comment]
Sorry, Seems I forgot to update the version in addon.xml.

Please try again:
https://github.com/bbsan2k/plugin.video.f1tv/releases/tag/0.1.5

The only change is the version string, so if you previously uninstalled 0.1.4 and reinstalled yesterdays' package there is no change for you.

Additionally make sure:
- your are not on VPN
- Your device list on f1tv is not maxed out

**Comment by dsgithub2k** [comment]
> Sorry, Seems I forgot to update the version in addon.xml.
> 
> Please try again:
> https://github.com/bbsan2k/plugin.video.f1tv/releases/tag/0.1.5
> 
> The only change is the version string, so if you previously uninstalled 0.1.4 and reinstalled yesterdays' package there is no change for you.
> 
> Additionally make sure:
> 
> * your are not on VPN
> * Your device list on f1tv is not maxed out

Hi @bbsan2k ,
So I uninstalled version 0.1.4, restarted Kodi, installed the new test 0.1.5 version and now it gives me a F1tv kodi plugin error message. Not on VPN and account on F1 is not maxed out, actually I deleted all devices for this test.  Reviewed log error on direct texture file loading failed, error getting plugin://plugin.video.f1tv/
error:control 55 in window 10025 has been asked to focus, but it can't 

Uninstalled and installed problem persists.

**Comment by bbsan2k** [comment]
@dsgithub2k Can you activate debug in the plugin and the log as well and provide the log output?

**Comment by LTKort** [comment]
Hi @bbsan2k , I got the following:

`2020-10-23 12:15:31.219 T:3011182608   ERROR: Control 55 in window 10025 has been asked to focus, but it can't
2020-10-23 12:15:31.693 T:2284041072   ERROR: en
2020-10-23 12:15:31.694 T:2284041072   ERROR: EXCEPTION Thrown (PythonToCppException) : -->Python callback/script returned the following error<--
                                             - NOTE: IGNORING THIS CAN LEAD TO MEMORY LEAKS!
                                            Error Type: <class 'locale.Error'>
                                            Error Contents: unsupported locale setting
                                            Traceback (most recent call last):
                                              File "/storage/.kodi/addons/plugin.video.f1tv/main.py", line 14, in <module>
                                                plugin.run()
                                              File "/storage/.kodi/addons/plugin.video.f1tv/resources/lib/plugin.py", line 530, in run
                                                router(sys.argv[2][1:])
                                              File "/storage/.kodi/addons/plugin.video.f1tv/resources/lib/plugin.py", line 504, in router
                                                get_mainpage()
                                              File "/storage/.kodi/addons/plugin.video.f1tv/resources/lib/plugin.py", line 41, in get_mainpage
                                                event_uid = _api_manager.getLiveEvent()
                                              File "/storage/.kodi/addons/plugin.video.f1tv/resources/lib/F1TVParser/F1TV_Minimal_API.py", line 134, in getLiveEvent
                                                elements = self.callAPI("home")['objects'][0]['items']
                                              File "/storage/.kodi/addons/plugin.video.f1tv/resources/lib/F1TVParser/F1TV_Minimal_API.py", line 47, in callAPI
                                                locale.setlocale(locale.LC_ALL, 'en_US')
                                              File "/usr/lib/python2.7/locale.py", line 581, in setlocale
                                            Error: unsupported locale setting
                                            -->End of Python script error report<--
2020-10-23 12:15:31.807 T:2444809072   ERROR: GetDirectory - Error getting plugin://plugin.video.f1tv/
2020-10-23 12:15:31.808 T:3011182608   ERROR: CGUIMediaWindow::GetDirectory(plugin://plugin.video.f1tv/) failed`

**Comment by LTKort** [comment]
@bbsan2k I Fixed it by commenting line #47 in F1TV_Minimal_API.py see #38

**Comment by Calimerorulez** [comment]
Same here. I can’t copy/paste the log right now, but:

ValueError: _locale emulation only supports “C” locale.

And: Control 55 in window 10025 has been asked to focus but it can’t

Error: nl 

Etc.

**Comment by bbsan2k** [comment]
@LTKort The problem is, that this only fixes the problem on Raspi and some other platforms, while on Mac and Windows it breaks functionality. I guess it has something to do with the way the cache lib is expecting the Date formats

Edit:
Merged for now. Please test 0.1.5.1

**Comment by dennis1804** [comment]
Thanks, 0.1.5.1 seems to work for android tv and Windows 😀
just like the comment below the current event is indeed missing.

**Comment by cmiguelcabral** [comment]
V0.1.5.1 working here on LibreElec (RPi3)
Just the current event is not appearing on the front page.

**Comment by Calimerorulez** [comment]
0.1.5.1 working for me too on Kodi/Chromecast with Google TV. 

Thanks all for the quick debugging and fixes!

**Comment by doormalena** [comment]
0.1.5.1 works on Windows 10 x64 (after manually installing it, repository does not know this version). Watching the Portugal P1 replay now.

It seems the navigation is changed though since the last release published in the repository, I can only find the current race via 2020 > Portugal. Not a blocker, just something to know.

**Comment by Calimerorulez** [comment]
> It seems the navigation is changed though since the last release published in the repository, I can only find the current race via 2020 > Portugal. Not a blocker, just something to know.

Same here

**Comment by dsgithub2k** [comment]
> 0.1.5.1 works on Windows 10 x64 (after manually installing it, repository does not know this version). Watching the Portugal P1 replay now.
> 
> It seems the navigation is changed though since the last release published in the repository, I can only find the current race via 2020 > Portugal. Not a blocker, just something to know.

I second that. Version 0.1.5.1 working for me on the Shield Tv Pro no error. The Navigational is changed, not sure if we'll be able to watch the live event?  Seems to show only past events.

**Comment by cmiguelcabral** [comment]
> > 0.1.5.1 works on Windows 10 x64 (after manually installing it, repository does not know this version). Watching the Portugal P1 replay now.
> > It seems the navigation is changed though since the last release published in the repository, I can only find the current race via 2020 > Portugal. Not a blocker, just something to know.
> 
> I second that. Version 0.1.5.1 working for me on the Shield Tv Pro no error. The Navigational is changed, not sure if we'll be able to watch the live event? Seems to show only past events.

It's playing live events, I watched P2 there.
Just the Current Event is not showing up on Root Page.

The update worked here, I installed this version having the previews installed and it showed "Updated..."

**Comment by dsgithub2k** [comment]
I concur with the above. Watched P2 today as well, although I joined in the second half of the live event. does the live event link appears right before is going to start?

**Comment by Calimerorulez** [comment]
Well does anyone know how to watch live?

**Comment by bbsan2k** [comment]
> Well does anyone know how to watch live?

Navigate to Portugal 2020 manually and the stream will appear once available.
Unfortunately it seems, live streaming only works on certain platforms - not including macOS, so it’s hard for me to debug.

**Comment by doormalena** [comment]
On Windows, the live stream works OK at the moment. Watching quali right now using 0.1.6 version.

**Comment by dsgithub2k** [comment]
App is working beautifully! Would like to thank the developer for making this happen Thank you @bbsan2k!

**Comment by adityakalra16** [comment]
Hey there. Installed the newest version, and yet still unable to play the race or quali videos. 
the highlights are playing but the rest are not.
Account details are correct and devices are all cleared, still, getting error in playing. Also not running any vpn.
Should i be running in case F1TV is not supported in my country? i tried after running vpn, it just doesnt open any session videos but at the same time no error logs. Please help! thank you

**Comment by bbsan2k** [comment]
> Hey there. Installed the newest version, and yet still unable to play the race or quali videos.
> the highlights are playing but the rest are not.
> Account details are correct and devices are all cleared, still, getting error in playing. Also not running any vpn.
> Should i be running in case F1TV is not supported in my country? i tried after running vpn, it just doesnt open any session videos but at the same time no error logs. Please help! thank you

Hey,
Which device/Kodi version is used?

**Comment by adityakalra16** [comment]
Hey! Thanks for the quick reply,
Currently using a Pi4 with LibreElec Kodi (not sure which version but the latest(fresh install today)

**Comment by Calimerorulez** [comment]
Same here. The only error I get is:

`ERROR: Account Authentication failed.`

Using Kodi 18.9 on a Chromecast Google TV. Couple of days ago, it was working fine.

**Comment by dsgithub2k** [comment]
I'm having no issues here. on Kodi/Nvidia Shield F1TV version 0.1.5

**Comment by Calimerorulez** [comment]
Still not working :'(

**Comment by doormalena** [comment]
> 
> 
> Still not working :'(

Having the same issue today with 0.1.6 when wanting to watch FP1. Previous race weekend it worked fine.

Using the official app with the same credentials is working.
Location: Netherlands

**Comment by dennis1804** [comment]
according to: https://github.com/SoMuchForSubtlety/f1viewer/issues/111.
it seems like the f1tv api is blocking bot calls because they can't execute javascript, which might be the reason why most of us can login.

**Comment by doormalena** [comment]
Interesting. According to further down in that thread it mentions its related to the user agent. See https://github.com/SoMuchForSubtlety/f1viewer/issues/111#issuecomment-726888417

**Comment by bbsan2k** [comment]
It is really difficult to reproduce this issue on my side. On my Mac, Live Events are not working - but they never have, so this is nothing "new".
Can somebody please try the latest version in develop and check if issues are resolved with this?
I added the User-Agent as "RaceControl" - and at least on my Raspberry Pi LibreElec it's working like this...

EDIT: Find Prelease 0.1.6.1 [here](https://github.com/bbsan2k/plugin.video.f1tv/releases/tag/0.1.6.1) @doormalena @dennis1804 @Calimerorulez

**Comment by dennis1804** [comment]
@bbsan2k, seems to work. (debain, ubuntu 20.04) however i am getting some errors when i try to watch the practice 3 of the turkish gp. other practices starting nicely from the beginning. 
```
11:08:24.053 T:140504246376192 WARNING: CRenderManager::WaitForBuffer - timeout waiting for buffer
11:08:30.344 T:140505669158656   ERROR: en
11:08:30.527 T:140505669158656   ERROR: EXCEPTION Thrown (PythonToCppException) : -->Python callback/script returned the following error<--
                                             - NOTE: IGNORING THIS CAN LEAD TO MEMORY LEAKS!
                                            Error Type: <type 'exceptions.TypeError'>
                                            Error Contents: 'NoneType' object has no attribute '__getitem__'
                                            Traceback (most recent call last):
                                              File "/home/dennis/.kodi/addons/plugin.video.f1tv/main.py", line 14, in <module>
                                                plugin.run()
                                              File "/home/dennis/.kodi/addons/plugin.video.f1tv/resources/lib/plugin.py", line 530, in run
                                                router(sys.argv[2][1:])
                                              File "/home/dennis/.kodi/addons/plugin.video.f1tv/resources/lib/plugin.py", line 495, in router
                                                playChannel(params['channel_url'])
                                              File "/home/dennis/.kodi/addons/plugin.video.f1tv/resources/lib/plugin.py", line 455, in playChannel
                                                stream_url= _api_manager.getChannelStream(channel_url=channel_url)
                                              File "/home/dennis/.kodi/addons/plugin.video.f1tv/resources/lib/F1TVParser/F1TV_Minimal_API.py", line 120, in getChannelStream
                                                return viewing_json["tokenised_url"]
                                            TypeError: 'NoneType' object has no attribute '__getitem__'
                                            -->End of Python script error report<--
11:08:30.587 T:140506978283456   ERROR: Playlist Player: skipping unplayable item: 0, path [plugin://plugin.video.f1tv/?action=playChannel&channel_url=%2Fapi%2Fchannels%2Fchan_7e1d189e283d469d81e07ba49dd11e80%2F]
```

**Comment by bbsan2k** [comment]
@dennis1804 This is probably related to the stream that has just finished, while not being available as recording yet... Probably you just need to wait a bit.
While still this is annoying (I agree) I currently don't have much time to invest into the plugin... I'll have some vacation before christmas, so maybe I can spend some time on making this whole thing a bit more robust...

**Comment by dennis1804** [comment]
yeah that's probably the issue, works now starting from the beginning. (also on android tv) thanks for making the time to fix these issues. :+1:

**Comment by merson316** [comment]
Can confirm. 0.1.6.1 has fixed it for me as well.

**Comment by Calimerorulez** [comment]
@bbsan2k 

Works OK for me too :) Thanks! (I was too late for checking live streaming though).

**Comment by doormalena** [comment]
> It is really difficult to reproduce this issue on my side. On my Mac, Live Events are not working - but they never have, so this is nothing "new".
> Can somebody please try the latest version in develop and check if issues are resolved with this?
> I added the User-Agent as "RaceControl" - and at least on my Raspberry Pi LibreElec it's working like this...
> 
> EDIT: Find Prelease 0.1.6.1 [here](https://github.com/bbsan2k/plugin.video.f1tv/releases/tag/0.1.6.1) @doormalena @dennis1804 @Calimerorulez

Just updated. Works fine with 0.1.6.1. Thanks for the quick fix.

**Comment by cmiguelcabral** [comment]
Just watched qualify live on version 1.6.0, everything good.

**Comment by Metalfreak82** [comment]
Had the same problem, there was a character in my password that didn't work I guess. Changed my password and can login now. Kodi on Android TV.

So if you have trouble logging in, try that.

**Comment by bjorn1981** [comment]
@bbsan2k  0.1.6.1 working on Kodi 18.8 Libreelec generic x86_64 from the Netherlands.

**Comment by bbsan2k** [comment]
Closed for now... Please open new issue if authentication starts failing again.
